neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+
+    Example:
+
+    >>> _decompose_unit(pq.mV)
+    ('volt', 0.001)
+    """
+    assert isinstance(unit, pq.quantity.Quantity)
+    assert unit.magnitude == 1
+    conversion = 1.0
+
+    def _decompose(unit):
+        dim = unit.dimensionality
+        if len(dim) != 1:
+            raise NotImplementedError("Compound units not yet supported")  # e.g. volt-metre
+        uq, n = list(dim.items())[0]
+        if n != 1:
+            raise NotImplementedError("Compound units not yet supported")  # e.g. volt^2
+        uq_def = uq.definition
+        return float(uq_def.magnitude), uq_def
+    conv, unit2 = _decompose(unit)
+    while conv != 1:
+        conversion *= conv
+        unit = unit2
+        conv, unit2 = _decompose(unit)
+    return list(unit.dimensionality.keys())[0].name, conversion
+
+
+def _recompose_unit(base_unit_name, conversion):
+    """
+    Given a base unit name and a conversion factor, return a quantities unit object
+
+    Example:
+
+    >>> _recompose_unit("ampere", 1e-9)
+    UnitCurrent('nanoampere', 0.001 * uA, 'nA')
+
+    """
+    if conversion not in prefix_map:
+        raise ValueError(f"Can't handle this conversion factor: {conversion}")
+    unit_name = prefix_map[conversion] + base_unit_name
+    if unit_name[-1] == "s":  # strip trailing 's', e.g. "volts" --> "volt"
+        unit_name = unit_name[:-1]
+    try:
+        return getattr(pq, unit_name)
+    except AttributeError:
+        logger.warning(f"Can't handle unit '{unit_name}'. Returning dimensionless")
+        return pq.dimensionless
+
+
+class NWBIO(BaseIO):
+    """
+    Class for "reading" experimental data from a .nwb file, and "writing" a .nwb file from Neo
+    """
+    supported_objects = [Block, Segment, AnalogSignal, IrregularlySampledSignal,
+                         SpikeTrain, Epoch, Event, ImageSequence]
+    readable_objects = supported_objects
+    writeable_objects = supported_objects
+
+    has_header = False
+    support_lazy = True
+
+    name = 'NeoNWB IO'
+    description = 'This IO reads/writes experimental data from/to an .nwb dataset'
+    extensions = ['nwb']
+    mode = 'one-file'
+
+    is_readable = True
+    is_writable = True
+    is_streameable = False
+
+    def __init__(self, filename, mode='r'):
+        """
+        Arguments:
+            filename : the filename
+        """
+        if not have_pynwb:
+            raise Exception("Please install the pynwb package to use NWBIO")
+        if not have_hdmf:
+            raise Exception("Please install the hdmf package to use NWBIO")
+        BaseIO.__init__(self, filename=filename)
+        self.filename = filename
+        self.blocks_written = 0
+        self.nwb_file_mode = mode
+
+    def read_all_blocks(self, lazy=False, **kwargs):
+        """
+
+        """
+        assert self.nwb_file_mode in ('r',)
+        io = pynwb.NWBHDF5IO(self.filename, mode=self.nwb_file_mode)  # Open a file with NWBHDF5IO
```

Comment:
```suggestion
        io = pynwb.NWBHDF5IO(self.filename, mode=self.nwb_file_mode, load_namespaces=True)  # Open a file with NWBHDF5IO
```

This will automatically generate the API for any extensions in the file.